python/pyarrow/tests/test_types.py

```diff
@@ -633,7 +643,7 @@ def test_struct_duplicate_field_names():
 def test_union_type():
     def check_fields(ty, fields):
         assert ty.num_fields == len(fields)
-        assert [ty[i] for i in range(ty.num_fields)] == fields
+        assert [ty.field(i) for i in range(ty.num_fields)] == fields
```

Comment:
Small last comment: can you keep both those lines? (we should still test that the `[]` version works as well)